### ISSUE TITLE
rvm_path setting not picking up

### ISSUE THREAD

**Issue by trustarun** [mina-deploy/mina#124]
My server have system wide install of rvm at /usr/local/rvm/scripts/rvm .

when I try to set it in my deploy.rb as below

set :rvm_path, "/usr/local/rvm/scripts/rvm"

it is not getting set and so when I try to deploy, it fail with below error message

 !     Ruby Version Manager not found
 !     If RVM is installed, check your :rvm_path setting.

I also tried with set_default
set_default :rvm_path, "/usr/local/rvm/scripts/rvm"

But problem is still there.

when I tried by hardcoding the path in below file of the gem it work smoothly:
https://github.com/nadarei/mina/blob/master/lib/mina/rvm.rb

Below is the modification:

original line:
set_default :rvm_path, "$HOME/.rvm/scripts/rvm"

hardcoded line by me:
set_default :rvm_path, "/usr/local/rvm/scripts/rvm"

So, now it work, but hardcoding the path is not a solution for me or any one else.

can you guide me what going wrong here....

I have seen the online documentation of all the possible settings ( http://nadarei.co/mina/settings/ ) and find that you do not list rvm_path there.

is the gem not supporting this  setting currently ..???

**Comment by helloyokoy** [comment]
thanks , use "set :rvm_path "
it works ~

**Comment by sintro** [comment]
Still have problem. The exact error message:
```
 mina setup                                                                                                                                 ⏎ master ✭   -       sh: 2: [[: not found
-       sh: 8: source: not found
-       sh: 9: rvm: not found
-----> Using RVM environment 'ruby-2.2.2-p95@default'

 !     Command failed.
       Failed with status 1 (1)
```

My deploy.rb:
```
require 'mina/bundler'
require 'mina/rails'
require 'mina/git'
# require 'mina/rbenv'  # for rbenv support. (http://rbenv.org)
require 'mina/rvm'    # for rvm support. (http://rvm.io)

# Basic settings:
#   domain       - The hostname to SSH to.
#   deploy_to    - Path to deploy into.
#   repository   - Git repo to clone from. (needed by mina/git)
#   branch       - Branch name to deploy. (needed by mina/git)

----- private info -----

# For system-wide RVM install.
set :rvm_path, '/usr/local/rvm/bin/rvm'

# Manually create these paths in shared/ (eg: shared/config/database.yml) in your server.
# They will be linked in the 'deploy:link_shared_paths' step.
set :shared_paths, ['config/database.yml', 'config/secrets.yml', 'log']

# Optional settings:
  set :user, 'deployer'    # Username in the server to SSH to.
#   set :port, '30000'     # SSH port number.
#   set :forward_agent, true     # SSH forward_agent.

# This task is the environment that is loaded for most commands, such as
# `mina deploy` or `mina rake`.
task :environment do
  # If you're using rbenv, use this to load the rbenv environment.
  # Be sure to commit your .ruby-version or .rbenv-version to your repository.
  # invoke :'rbenv:load'

  # For those using RVM, use this to load an RVM version@gemset.
  invoke :'rvm:use[ruby-2.2.2-p95@default]'
end

# Put any custom mkdir's in here for when `mina setup` is ran.
# For Rails apps, we'll make some of the shared paths that are shared between
# all releases.
task :setup => :environment do
  queue! %[mkdir -p "#{deploy_to}/#{shared_path}/log"]
  queue! %[chmod g+rx,u+rwx "#{deploy_to}/#{shared_path}/log"]

  queue! %[mkdir -p "#{deploy_to}/#{shared_path}/config"]
  queue! %[chmod g+rx,u+rwx "#{deploy_to}/#{shared_path}/config"]

  queue! %[touch "#{deploy_to}/#{shared_path}/config/database.yml"]
  queue! %[touch "#{deploy_to}/#{shared_path}/config/secrets.yml"]
  queue  %[echo "-----> Be sure to edit '#{deploy_to}/#{shared_path}/config/database.yml' and 'secrets.yml'."]

  if repository
    repo_host = repository.split(%r{@|://}).last.split(%r{:|\/}).first
    repo_port = /:([0-9]+)/.match(repository) && /:([0-9]+)/.match(repository)[1] || '22'

    queue %[
      if ! ssh-keygen -H  -F #{repo_host} &>/dev/null; then
        ssh-keyscan -t rsa -p #{repo_port} -H #{repo_host} >> ~/.ssh/known_hosts
      fi
    ]
  end
end

desc "Deploys the current version to the server."
task :deploy => :environment do
  to :before_hook do
    # Put things to run locally before ssh
  end
  deploy do
    # Put things that will set up an empty directory into a fully set-up
    # instance of your project.
    invoke :'git:clone'
    invoke :'deploy:link_shared_paths'
    invoke :'bundle:install'
    invoke :'rails:db_migrate'
    invoke :'rails:assets_precompile'
    invoke :'deploy:cleanup'

    to :launch do
      queue "mkdir -p #{deploy_to}/#{current_path}/tmp/"
      queue "touch #{deploy_to}/#{current_path}/tmp/restart.txt"
    end
  end
end

# For help in making your deploy script, see the Mina documentation:
#
#  - http://nadarei.co/mina
#  - http://nadarei.co/mina/tasks
#  - http://nadarei.co/mina/settings
#  - http://nadarei.co/mina/helpers
```

What I did is changed user to 'deployer' and changed rvm_path
```
set :rvm_path, '/usr/local/rvm/bin/rvm'
```
If I just connect through terminal to ssh as deployer, rvm works fine.

PS I am using rbenv on local machine, I hope I don`t need to have rvm on local for correct mina operation?

**Comment by AlfredoRoca** [comment]
I don't see your repository var:
set :repository, 'git@...

**Comment by sintro** [comment]
I have cut that in '----- private info -----'.

Actually, I found the source of half of my problems. It was default system shell of Ubuntu, which I use as server OS. It has `dash`, not `bash`. The point is `dash` even don't have 'source' command, which is heavily used in mina. Dash has '.' command in place of 'source'. 
So I changed dash to bash as default shell and all the weird errors are gone... But appeared new, more common errors with RVM. Mina deploy just can not see it. Mina SSH too. 
For now i temporarily solved the issue by manually sourcing all needed env vars and paths for my Ruby. Mina`s rvm package could not help me in this case. Hope this is problem with Ubuntu, not Mina.
